### DISCUSSION TITLE
MetaData from local MP3 files

### DISCUSSION THREAD

**Discussion by FrankJaco** [music-assistant/support#1314] [Q&A] [answered]
MASS does a great job of obtaining Metadata such as the Album and Artist thumbnails from Music Brainz and other providers. However I have a sizable amount of local files where no Metadata exists online. These all appear without icons in the MASS UI (other than their initials). But I have embedded (properly I believe) the thumbnails and other data in my MP3 files.  Why is it that MASS doesn't display these when the Metadata services do not find any?  I believe my files are correctly formed as other services and Apps (e.g. Plex, Pulsar) display my thumbnails etc.

Does MASS default to local MetaData when no match is available online?  If not, I will make this a feature request.

Thanks,
fj

**Comment by OzGav** [comment]
From Discord 

> for local files, you can have an artist folder with the artist.nfo in there (just like the images) and that will be preferred. Online metadata providers are only queried when there is no local data.

https://kodi.wiki/view/NFO_files

**Reply by marcelveldt** [reply, marked as answer]
Local metadata is ALWAYS preferred over online metadata:
- artwork embedded in the media file
- image files within the artist/album folder structure (e.f. folder.jpg, disc.png etc. etc)
- additional metadata such as descriptions from nfo files

If this does not work, its a bug.

Online metadata is only scraped when we got nothing local

**Reply by kreativkod** [reply]
> Local metadata is ALWAYS preferred over online metadata:
> 
>     * artwork embedded in the media file
> 
>     * image files within the artist/album folder structure (e.f. folder.jpg, disc.png etc. etc)
> 
>     * additional metadata such as descriptions from nfo files
> 
> 
> If this does not work, its a bug.
> 
> Online metadata is only scraped when we got nothing local

Does this also apply to ID3 TAGS? Can I completely skip the step of matching my carefully curated flac collection of 70 000 files with online sources, just using my ID3 tags and **_front_**.jpg alongside the files?

**Reply by OzGav** [reply]
I am not sure what “step” you are referring to but if these tags are incomplete, particularly with regard to the unique Musicbrainz tags, then you will likely have a poor experience.

**Reply by kreativkod** [reply]
> I am not sure what “step” you are referring to but if these tags are incomplete, particularly with regard to the unique Musicbrainz tags, then you will likely have a poor experience.

"Step" was probably a badly chosen word. I was trying to refer to whenever MASS looks at my NAS files and integrates them into the unified library. I haven't installed yet, trying to gauge whether the MASS+HA route is for me or if I should stay on my legacy RuneAudio+BubbleUPnP for Qobuz integration-system.

So the unique MusicBrainz tags in ID3 are important to organizing local files, MASS is not just calling an album whatever is entered in the %album% tag for instance? Does it prioritize e.g. MusicBrainz Release Id over %album% and MusicBrainz Artist Id or MusicBrainz Releaseartist Id over %artist%/%albumartist%? 

I have a huge mass of classical music files and MusicBrainz, for all its wonders, does not make a good job of harmonizing them into a usable library.

**Reply by OzGav** [reply]
MA will take the basic tags if it has to and will appropriately add those to the library. However, one of MA's features it will link the same tracks together and collapse them into one. If you have tracks or albums that are named the same but are different without unique identifiers to discriminate them then you will likely get incorrect matches. If you read this https://discord.com/channels/753947050995089438/1276425571689037837/1276427287331475487 you will get some good info from someone else who is working on getting their classical music into MA.
